model/project.go

```diff
@@ -1231,6 +1231,32 @@ func (p *Project) FindTaskGroup(name string) *TaskGroup {
 	return nil
 }
 
+// FindTaskGroupForTask returns a specific task group from a project
+// that contains the given task.
+func (p *Project) FindTaskGroupForTask(name string) *TaskGroup {
+	for _, tg := range p.TaskGroups {
+		for _, t := range tg.Tasks {
+			if t == name {
+				return &tg
+			}
+		}
+	}
+
+	for _, bv := range p.BuildVariants {
```

Comment:
I believe inline task groups take precedence over the top-level defined task groups, so these would have to be checked first.

Reply:
I revamped it to first get a mapping of all the potential task groups it is a part of, then go through the inline then go through the potential mapping to see what it might go to.